### DISCUSSION TITLE
Interaction testing router events

### DISCUSSION THREAD

**Discussion by joemcilwraith1107** [RyanClementsHax/storybook-addon-next#147] [Q&A] [answered]
WIth storybook you can use the interactions addon to test events (like onCLick) that are sent to the actions panel by setting argTypes as set out in the  [actions addon docs](https://storybook.js.org/docs/react/essentials/actions). If the onCLick event is set via an arg called onclick then you can test that it gets called by doing something like (if you're clicking an Button that contains the text `Edit`

```
await userEvent.click(canvas.getByText("Edit"));
await waitFor(() => expect(args.onClick).toHaveBeenCalled())
```

Given that this addon sends Link clicks (for example) to the actions panel, is there a way to check that those events are called?

**Comment by RyanClementsHax** [comment, marked as answer]
This question is probably best asked over on the storybook side. I'm unfortunately unfamiliar with storybook actions. In addition, storybook is currently baking in nextjs framework support themselves so this would be good feedback to provide them.
